OptimizelySDK/Optimizely.cs

```diff
@@ -153,19 +153,7 @@ public string Activate(string experimentKey, string userId, UserAttributes userA
                 userAttributes = userAttributes.FilterNullValues(Logger);
             }
 
-            var impressionEvent = EventBuilder.CreateImpressionEvent(Config, experiment, variation.Id, userId, userAttributes);
-            Logger.Log(LogLevel.INFO, string.Format("Activating user {0} in experiment {1}.", userId, experimentKey));
-            Logger.Log(LogLevel.DEBUG, string.Format("Dispatching impression event to URL {0} with params {1}.", 
-                impressionEvent, impressionEvent.GetParamsAsJson()));
-
-            try
-            {
-                EventDispatcher.DispatchEvent(impressionEvent);
-            }
-            catch (Exception exception)
-            {
-                Logger.Log(LogLevel.ERROR, string.Format("Unable to dispatch impression event. Error {0}", exception.Message));
-            }
+            SendImpressionEvent(experiment, variation, userId, userAttributes);
```

Comment:
Passing variation.id, as done earlier was better since you do not need variation object in impression method. Csharp, by default, passes params by value so less memory alloc.

Reply:
Replaced Variation object with Variation ID in SendImpressionEvent method. 

Btw, unless specified using ref or out parameters, C# pass references of non primitive types by value.